src/main/java/org/candlepin/policy/js/entitlement/EntitlementRules.java

```diff
@@ -749,16 +749,19 @@ private PoolOperationCallback adjustHostedBonusPoolQuantity(PoolManager poolMana
                         notConsumedLocally += entitlement.getQuantity();
                     }
 
-                    if (pool.getQuantity().equals(notConsumedLocally)) {
-                        // getting all pools matching the sub id. Filtering out
-                        // the 'parent'.
-                        List<Pool> pools = subscriptionPoolMap.get(pool.getSubscriptionId());
-                        if (pools != null) {
-                            for (int idex = 0; idex < pools.size(); idex++) {
-                                Pool derivedPool = pools.get(idex);
-                                if (derivedPool.getAttributeValue(Pool.Attributes.DERIVED_POOL) != null) {
+                    // getting all pools matching the sub id. Filtering out
+                    // the 'parent'.
+                    List<Pool> pools = subscriptionPoolMap.get(pool.getSubscriptionId());
+                    if (pools != null) {
+                        for (int idex = 0; idex < pools.size(); idex++) {
```

Comment:
Let's modernize this loop and address the piles of nesting. Either streams or a for-each loop would clean it up substantially. For example:
```
pools.stream()
    .filter(pool -> pool.getAttributeValue(Pool.Attributes.DERIVED_POOL) != null)
    .forEach(pool -> poolOperationCallback.setQuantityToPool(pool, 
        pool.getQuantity().equals(notConsumerLocally) ? 0 : -1));
```